bundler/lib/bundler/definition.rb

```diff
@@ -339,7 +339,10 @@ def lock(file, preserve_unknown_sections = false)
 
       preserve_unknown_sections ||= !updating_major && (Bundler.frozen_bundle? || !(unlocking? || @unlocking_bundler))
 
-      return if file && File.exist?(file) && lockfiles_equal?(@lockfile_contents, contents, preserve_unknown_sections)
+      if file && File.exist?(file) && lockfiles_equal?(@lockfile_contents, contents, preserve_unknown_sections)
+        SharedHelpers.filesystem_access(file) {|p| FileUtils.touch(p) }
+        return
```

Comment:
I would specifically return without touching the lockfile if the bundle is frozen, like so:

```suggestion
        return if Bundler.frozen_bundle?
        SharedHelpers.filesystem_access(file) {|p| FileUtils.touch(p) }
        return
```

Reply:
I'd likewise add a test that it is not touched when frozen. Maybe [here](https://github.com/rubygems/rubygems/blob/master/bundler/spec/install/deploy_spec.rb#L262), since the lock command always temporarily unfreezes.